larpandoracontent/LArHelpers/LArFormattingHelper.cc

```diff
@@ -0,0 +1,184 @@
+    
+    void LArFormattingHelper::Table::UpdateColumnWidth()
+    {
+        const unsigned int currentElementIndex(m_elements.size()-1);
+        const unsigned int column(currentElementIndex % m_widths.size());
+        m_widths[column] = std::max(static_cast<unsigned int>(m_elements[currentElementIndex].length()), m_widths[column]);
+    }
+    
+    //--------------------------------------------------------------------------------------------------------------------------------------
+   
+    void LArFormattingHelper::Table::Print()
+    {
+        if (m_columnTitles.size() == 0)
+            return;
+
+        this->PrintColumnTitles();
+        this->PrintHorizontalLine();
+        this->PrintTableElements();
+    }
+
+    //--------------------------------------------------------------------------------------------------------------------------------------
+    
+    void LArFormattingHelper::Table::PrintColumnTitles()
+    {
+        for (unsigned int i=0; i<m_columnTitles.size(); i++)
+            this->PrintTableCell(m_columnTitles[i], LArFormattingHelper::GetFormatCharacter(LArFormattingHelper::BOLD), i);
+    }
+    
+    //--------------------------------------------------------------------------------------------------------------------------------------
+    
+    void LArFormattingHelper::Table::PrintHorizontalLine()
+    {
+        for (unsigned int i=0; i<m_columnTitles.size(); i++) 
+            this->PrintTableCell(std::string(m_widths[i], '-'), LArFormattingHelper::GetFormatCharacter(LArFormattingHelper::REGULAR), i);
+    }
+
+    //--------------------------------------------------------------------------------------------------------------------------------------
+    
+    void LArFormattingHelper::Table::PrintTableElements()
+    {
+        const unsigned int nRows((m_elements.size() + m_columnTitles.size() - 1)/m_columnTitles.size());
+        if (nRows*m_columnTitles.size() != m_elements.size())
+        {
+            std::cerr << "LArFormattingHelper::Table::PrintTableElements - Error: Number of table elements added doesn't fill a whole row" << std::endl;
+            throw StatusCodeException(STATUS_CODE_OUT_OF_RANGE);
+        }
+
+        for (unsigned int i=0; i<nRows*m_columnTitles.size(); i++)
+            this->PrintTableCell(m_elements[i], m_format[i], i);
+    }
+    
+    //--------------------------------------------------------------------------------------------------------------------------------------
+    
+    void LArFormattingHelper::Table::PrintTableCell(const std::string &value, const std::string &format, const unsigned int &index)
+    {
+        const unsigned int column(index % m_columnTitles.size());
+
+        std::string separator(this->IsSeparatorColumn(column) ? "" : " ");
+        std::cout << "|" << format << separator << std::setw(m_widths[column]) << std::internal << value << separator;
+        LArFormattingHelper::ResetStyle();
+
+        if (column == m_columnTitles.size()-1)
```

Comment:
Space either side of minus sign; what happens if m_columnTitles is empty?

Reply:
Added check for empty `m_columnTitles` for safety, although this function is only called by functions if `m_columnTitles` is not empty